server/src/game/cards/10-experimental/HeroJealous.ts

```diff
@@ -0,0 +1,51 @@
+import CardType from '@shared/enums/CardType'
+import CardColor from '@shared/enums/CardColor'
+import CardFaction from '@shared/enums/CardFaction'
+import GameEventType from '@shared/enums/GameEventType'
+import CardFeature from '@shared/enums/CardFeature'
+import ExpansionSet from '@shared/enums/ExpansionSet'
+import TargetType from '@shared/enums/TargetType'
+
+import ServerPlayerInGame from '@src/game/players/ServerPlayerInGame'
+import ServerAnimation from '@src/game/models/ServerAnimation'
+import ServerCard from '@src/game/models/ServerCard'
+import ServerGame from '@src/game/models/ServerGame'
+
+export default class HeroJealous extends ServerCard {
+	constructor(game: ServerGame) {
+		super(game, {
+			type: CardType.UNIT,
+			color: CardColor.GOLDEN,
+			faction: CardFaction.NEUTRAL,
+			features: [CardFeature.KEYWORD_DEPLOY],
+			stats: {
+				power: 1,
+			},
+			expansionSet: ExpansionSet.BASE,
+			isExperimental: true,
+		})
+
+		this.createDeployEffectTargets()
+			.totalTargets(1)
+			.target(TargetType.CARD_IN_LIBRARY)
+			.require(TargetType.CARD_IN_LIBRARY, ({ targetCard }) => targetCard.type === CardType.UNIT)
+
+		this.createEffect(GameEventType.CARD_TARGET_SELECTED_CARD)
+			.perform(({ targetCard }) => this.onTargetSelected(targetCard))
+			
+		// TODO On death: remove from game
+	}
+
+	private onTargetSelected(target: ServerCard): void {
+		let affected: ServerCard[] = []
+		this.game.players.forEach((player) => affected.concat(discardCopiesFromHand(target, player)))
+		this.game.animation.play(ServerAnimation.cardAffectsCards(this, affected))
```

Comment:
You are playing animation for cards that are no longer in play. Or at least they are now in graveyard. As both the deck and the graveyard don't have graphical representations of cards, there will be no animation at all. You should be playing it before discarding the card.

Also you can start animation thread for every discard and play an animation for that one card inside the thread.